patroni/config.py

```diff
@@ -601,3 +602,12 @@ def get_global_config(self, cluster: Union[Cluster, None]) -> GlobalConfig:
         :returns: :class:`GlobalConfig` object
         """
         return get_global_config(cluster, self._dynamic_configuration)
+
+    def _validate_failover_tags(self):
+        """Ensures that only one of `no_failover` and `failover_priority` is set"""
+        has_nofailover_tag = self.get('tags', {}).get('nofailover', None) is not None
+        has_failover_priority_tag = self.get('tags', {}).get('failover_priority', None) is not None
+        if has_nofailover_tag and has_failover_priority_tag:
+            raise ConfigParseError(
+                'Cannot provide values for both `nofailover` and `failover_priority`. Please set only one.'
+            )
```

Comment:
Why do we need to have this restriction?

Reply:
The way I have it set up right now, `failover_priority: 0` means "Never pursue leadership". This means that having `failover_priority: 0` and `nofailover: false` would be an ambiguous and invalid configuration. I thought it might be simpler to just force only one of these to be provided, since anytime you specify a `failover_priority` you are requiring a specific `nofailover` value.

I could change it so that it validates if the configuration is consistent between the two? Or other thoughts on how to rectify this? (I imagine we still want `nofailover` to have the exact same impact for backwards compatibility)